src/mainboard/starfive/visionfive2/bt0/src/main.rs

```diff
@@ -225,140 +277,85 @@ fn main() {
-        let size = 0x0020_0000; // occupied space
-        let dram = DRAM_BASE;
-        // let's find the dtb
-
-        let slice = unsafe {
-            let pointer = transmute(SRAM0_BASE);
-            // The `slice` function creates a slice from the pointer.
-            unsafe { core::slice::from_raw_parts(pointer, size) }
-        };
-        let fdt = find_fdt(slice);
-        match fdt {
-            Err(_) => {
-                println!(
-                    "Could not find an FDT between {SRAM0_BASE:08x} and {:08x}",
-                    SRAM0_BASE + size
-                );
-            }
-            Ok(f) => {
-                dtb = SRAM0_BASE + 0x10000; // unsafe {(&f as *const _ as usize)};
-                let it = &mut f.find_all_nodes("/flash-info/areas");
-                let a = FdtIterator::new(it);
-                for aa in a {
-                    for c in aa.children() {
-                        for p in c.properties() {
-                            match p.name {
-                                "size" => {
-                                    println!("{:?} / {:?}, {:?}", c.name, p.name, p.as_usize());
-                                }
-                                _ => {
-                                    println!("{:?} / {:?} {:?}", c.name, p.name, p.as_str());
-                                }
-                            }
-                        }
-                    }
-                }
-            }
-        }
-
-        for b in (0..size).step_by(4) {
-            write32(dram + b, read32(base + b));
-            if b % 0x4_0000 == 0 {
-                print!(".");
-            }
-        }
-        let size = 0x0010_0000; // occupied space
-        let base = QSPI_XIP_BASE + 0x0030_00d4; // first 0xd4 is just 0-bytes
-        let target = DRAM_BASE + 0x0020_0000;
-        for b in (0..size).step_by(4) {
-            write32(target + b, read32(base + b));
-            if b % 0x4_0000 == 0 {
-                print!(".");
-            }
-        }
-        println!(" done.");
-        if DEBUG {
-            println!("Start:");
-            dump_block(dram, 0x100, 0x20);
-            println!("Presumably JH7110 recovery:");
-            dump_block(dram + 0x0002_0000, 0x100, 0x20);
-            println!("DTB:");
-            dump_block(dram + 0x0010_0000, 0x100, 0x20);
-            println!("Something:");
-            dump_block(dram + 0x0020_0000, 0x100, 0x20);
-        }
+    // Find and copy the main stage
+    let (base, size) = if LOAD_FROM_FLASH {
+        (QSPI_XIP_BASE, FLASH_SIZE)
     } else {
-        let base = 0x0800_0000 + 32 * 1024;
-        let dram = DRAM_BASE;
-        for b in (0..0x100).step_by(4) {
-            write32(dram + b, read32(base + b));
-            if b % 0x4_0000 == 0 {
-                print!(".");
-            }
+        (SRAM0_BASE, SRAM0_SIZE) // occupied space
+    };
+    let slice = unsafe { core::slice::from_raw_parts(transmute(base), size) };
```

Comment:
Maybe add a `SAFETY` comment here since we are loading executable memory from FLASH/RAM?